### PR TITLE
CAL-10 Moved nitf input transformer here, adjusted pom files accordin…

### PR DESCRIPTION
**Pull request by beyelerb**
#### What does this PR do?
Moved the nitf input transformer into the alliance repo. Made it part of the overall build.
#### Who is reviewing it?
@jlcsmith 
@kcwire 
@dcruver 
@jrnorth 
#### How should this be tested?
Drop bundle into DDF, use upload function to exercise, view created metadata in the UI.
#### What are the relevant tickets?
CAL-10
#### Checklist:
- [ ] Documentation Updated
- [X ] Update / Add Unit Tests
- [X ] Update / Add Integration Tests

…gly.

### CODE REVIEW

File: catalog/imaging/transformer/catalog-transformer-nitf/pom.xml [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Embed-Dependency>
+                            catalog-core-api-impl,
+                            catalog-core-dynamic-impl,
+                            platform-util,
+                            commons-beanutils,
+                            cdi-api,
+                            commons-beanutils,
```
Comment (jrnorth) [inline catalog/imaging/transformer/catalog-transformer-nitf/pom.xml]: Embedded twice.

File: catalog/imaging/transformer/catalog-transformer-nitf/src/main/resources/nitf.xml [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+        <!--addMultivaluedDescriptor(FILE_TITLE, String.class, true, true, false);-->
+        <attribute>
+            <name>fileTitle</name>
+            <type>String</type>
+            <indexed>true</indexed>
+            <stored>true</stored>
+            <tokenized>false</tokenized>
+            <multi-valued>true</multi-valued>
+        </attribute>
+        <!--addMultivaluedDescriptor(FILE_SIZE, Long.class, true, true, false);-->
```
Comment (jrnorth) [inline catalog/imaging/transformer/catalog-transformer-nitf/src/main/resources/nitf.xml]: I think some of these comments incorrectly say "addMultivaluedDescriptor" when the attribute isn't multi-valued.

File: catalog/imaging/transformer/catalog-transformer-nitf/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java [hunk 3]
```diff
@@ -0,0 +1,667 @@
...
+        metacard.setAttribute(Metacard.MODIFIED, date);
+        metacard.setAttribute(Metacard.CREATED, date);
+        metacard.setAttribute(Metacard.EFFECTIVE, date);
+
+        metacard.addAttribute(Nitf.FILE_TITLE, header.getFileTitle());
+        metacard.addAttribute(Nitf.COMPLEXITY_LEVEL, header.getComplexityLevel());
+        metacard.addAttribute(Nitf.ORIGINATOR_NAME, header.getOriginatorsName());
+        metacard.addAttribute(Nitf.ORIGINATING_STATION_ID, header.getOriginatingStationId());
+
+        metacard.addAttribute(Nitf.CODE_WORDS,
```
Comment (jrnorth) [inline catalog/imaging/transformer/catalog-transformer-nitf/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java]: Might want to put the object returned by `getFileSecurityMetadata()` into a local variable.

File: catalog/imaging/transformer/catalog-transformer-nitf/src/test/java/ddf/catalog/transformer/nitf/TestNitfInputTransformer.java [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+    private static final String GEO_NITF = "/i_3001a.ntf";
+
+    private static MetacardFactory metacardFactory = null;
+
+    private NitfInputTransformer transformer = null;
+
+    @BeforeClass
+    public static void setupBeforeClass() throws Exception {
+        MetacardFactory mf = new MetacardFactoryImpl();
```
Comment (jrnorth) [inline catalog/imaging/transformer/catalog-transformer-nitf/src/test/java/ddf/catalog/transformer/nitf/TestNitfInputTransformer.java]: This local `MetacardFactory` isn't necessary - you could just work with `metacardFactory`.

**Comment by jlcsmith** [comment]
global comment: we need tickets for the todo's here

File: catalog/imaging/transformer/catalog-transformer-nitf/src/test/java/ddf/catalog/transformer/nitf/TestNitfInputTransformer.java [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+        MetacardTypeReader reader = new MetacardTypeReader();
+        reader.setMetacardFactory(mf);
+        InputStream is = Thread.currentThread()
+                .getContextClassLoader()
+                .getResourceAsStream("nitf.xml");
+        reader.parseMetacardDefinition(is);
+        metacardFactory = mf;
+        System.setProperty("java.awt.headless", "true");
+        Toolkit tk = Toolkit.getDefaultToolkit();
```
Comment (jrnorth) [inline catalog/imaging/transformer/catalog-transformer-nitf/src/test/java/ddf/catalog/transformer/nitf/TestNitfInputTransformer.java]: Unused?

File: catalog/imaging/transformer/catalog-transformer-nitf/src/test/java/ddf/catalog/transformer/nitf/TestNitfInputTransformer.java [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    }
+
+    @Before
+    public void createTransformer()
+            throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+        transformer = new NitfInputTransformer();
+        CatalogFramework catalog = mock(CatalogFramework.class);
```
Comment (jrnorth) [inline catalog/imaging/transformer/catalog-transformer-nitf/src/test/java/ddf/catalog/transformer/nitf/TestNitfInputTransformer.java]: Unused?

File: catalog/imaging/transformer/catalog-transformer-nitf/src/test/java/ddf/catalog/transformer/nitf/TestNitfInputTransformer.java [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+            SourceUnavailableException, FederationException, ParseException {
+        Metacard metacard = transformer.transform(getInputStream(GEO_NITF));
+
+        assertNotNull(metacard);
+
+        validateDate(metacard, metacard.getCreatedDate(), "1997-12-17 10:26:30");
+        validateDate(metacard, metacard.getEffectiveDate(), "1997-12-17 10:26:30");
+        validateDate(metacard, metacard.getModifiedDate(), "1997-12-17 10:26:30");
+        //validateDate(metacard, metacard.getExpirationDate(), "1997-12-17 10:26:30");
+        assertThat(Nitf.NAME, is(metacard.getMetacardType().getName()));
```
Comment (jrnorth) [inline catalog/imaging/transformer/catalog-transformer-nitf/src/test/java/ddf/catalog/transformer/nitf/TestNitfInputTransformer.java]: I think it makes more sense to use the pattern `assertThat(valueToTest, is(expectedValue))`.

File: catalog/imaging/transformer/catalog-transformer-nitf/src/test/java/ddf/catalog/transformer/nitf/TestNitfInputTransformer.java [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+        //assertThat("", is(metacard.getResourceURI()));
+        //assertThat("", is(metacard.getSourceId()));
+        assertThat(
+                "Checks an uncompressed 1024x1024 8 bit mono image with GEOcentric data. Airfield",
+                is(metacard.getTitle()));
+        String wkt = metacard.getLocation();
+        assertTrue(wkt.matches(
+                "^POLYGON \\(\\(85 32.98\\d*, 85.00\\d* 32.98\\d*, 85.00\\d* 32.98\\d*, 85 32.98\\d*, 85 32.98\\d*\\)\\)"));
+
+        assertThat(8, equalTo(metacard.getAttribute(Nitf.BITS_PER_PIXEL_PER_BAND)
```
Comment (jrnorth) [inline catalog/imaging/transformer/catalog-transformer-nitf/src/test/java/ddf/catalog/transformer/nitf/TestNitfInputTransformer.java]: It's probably best to pick either `equalTo` or `is` and stick with it.

**Comment by jlcsmith** [comment]
I think we should remove the metadata creation once DDF-1396 is merged.

File: catalog/imaging/transformer/catalog-transformer-nitf/pom.xml [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Embed-Dependency>
+                            catalog-core-api-impl,
+                            catalog-core-dynamic-impl,
+                            platform-util,
+                            commons-beanutils,
+                            cdi-api,
```
Comment (jrnorth) [inline catalog/imaging/transformer/catalog-transformer-nitf/pom.xml]: I'm curious - I don't see `cdi-api` or `codice-imaging-cgm` explicitly listed as dependencies anywhere. Don't you need `<Embed-Transitive>true</Embed-Transitive>` to embed transitive dependencies like that?

File: catalog/imaging/transformer/catalog-transformer-nitf/src/main/java/ddf/catalog/transformer/nitf/Nitf.java [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+
+    String CLASSIFICATION_REASON = "reason";
+
+    String CLASSIFICATION_DATE = "classificationDate";
+
+    String DECLASSIFICATION_TYPE = "declassificationType";
+
+    String DECLASSIFICATION_DATE = "declassificationDate";
+
+    String SECURITY = "";
```
Comment (jrnorth) [inline catalog/imaging/transformer/catalog-transformer-nitf/src/main/java/ddf/catalog/transformer/nitf/Nitf.java]: Is this supposed to be empty?

File: catalog/imaging/transformer/catalog-transformer-nitf/src/test/java/ddf/catalog/transformer/nitf/TestNitfInputTransformer.java [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+    @BeforeClass
+    public static void setupBeforeClass() throws Exception {
+        metacardFactory = new MetacardFactoryImpl();
+        MetacardTypeReader reader = new MetacardTypeReader();
+        reader.setMetacardFactory(metacardFactory);
+        InputStream is = Thread.currentThread()
+                .getContextClassLoader()
+                .getResourceAsStream("nitf.xml");
+        reader.parseMetacardDefinition(is);
+        System.setProperty("java.awt.headless", "true");
```
Comment (jrnorth) [inline catalog/imaging/transformer/catalog-transformer-nitf/src/test/java/ddf/catalog/transformer/nitf/TestNitfInputTransformer.java]: Should this property be reset after the tests are done?

**Comment by beyelerb** [comment]
This pr has been included in the work @dcruver is doing. Closing this one and moving any comments not already addressed to his PR.

**Pull request closed without merging**